src/main/java/io/vertx/openapi/contract/OpenAPIVersion.java

```diff
@@ -75,16 +76,42 @@ public static OpenAPIVersion fromContract(JsonObject contract) {
     }
   }
 
-  public Future<OutputUnit> validate(Vertx vertx, SchemaRepository repo, JsonObject contract) {
-    return vertx.executeBlocking(p -> p.complete(repo.validator(mainSchemaFile).validate(contract)));
+  public Future<OutputUnit> validateContract(Vertx vertx, SchemaRepository repo, JsonObject contract) {
+    return vertx.executeBlocking(() -> repo.validator(mainSchemaFile).validate(contract));
+  }
+
+  /**
+   * Validates additional contract files against the openapi schema. If validations fails, try to validate against the
+   * json schema specifications only.
+   *
+   * @param vertx                   The related Vert.x instance.
+   * @param repo                    The SchemaRepository to do the validations with.
+   * @param file                    The additional json contract to validate.
+   */
+  public Future<Void> validateAdditionalContractFiles(Vertx vertx, SchemaRepository repo, JsonObject file) {
+    return validateContract(vertx, repo, file)
```

Comment:
Why did this method also validate the contract?

Reply:
It first tries to validate it against the openapi scheme first. Maybe this is too much validations and we can just have it validate against the OpenApi version's draft schemes instead?

Reply:
I've decided to remove validation against the openapi spec as that is really just a super set of the json schema spec. The additional files only really need to be a valid json spec.